<launch>

  <!-- args corresponding to environment variables -->
  <arg name="UAV_NAME" default="$(env UAV_NAME)" />
  <arg name="PROFILER" default="$(optenv PROFILER false)" />

  <!-- other args -->
  <arg name="standalone" default="true" />
  <arg name="debug" default="false" />

  <!-- frame ids -->
  <arg name="fcu_frame_id" default="fcu" />
  <arg name="fcu_untilted_frame_id" default="fcu_untilted" />

  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="standalone" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet" value="load" />
  <arg     if="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="" />
  <arg unless="$(eval arg('standalone') or arg('debug'))" name="nodelet_manager" value="$(arg UAV_NAME)_nodelet_manager" />

  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />

  <group ns="$(arg UAV_NAME)">

    <node pkg="nodelet" type="nodelet" name="transform_manager" args="$(arg nodelet) mrs_uav_state_estimation/TransformManager $(arg nodelet_manager)" output="screen" launch-prefix="$(arg launch_prefix)">

      <param name="enable_profiler" value="$(arg PROFILER)" />
      <param name="uav_name" value="$(arg UAV_NAME)" />
      <param name="frame_id/fcu" value="$(arg fcu_frame_id)" />
      <param name="frame_id/fcu_untilted" value="$(arg fcu_untilted_frame_id)" />

      <rosparam file="$(find mrs_uav_state_estimation)/config/transform_manager.yaml" command="load" />
      <rosparam file="$(find mrs_uav_state_estimation)/config/estimation_manager.yaml" command="load" />

      <!-- Subscribers -->
      <remap from="~mavros_odom_in" to="mavros/local_position/odom" />

      <!-- Publishers -->
      <remap from="~profiler" to="profiler" />

    </node>

  </group>

</launch>
